node/network/approval-distribution/src/lib.rs

```diff
@@ -990,6 +1174,14 @@ impl State {
 		let message_subject = MessageSubject(block_hash, candidate_index, validator_index);
 		let message_kind = MessageKind::Approval;
 
+		let peer_id = source.peer_id();
+		gum::debug!(
+			target: LOG_TARGET,
+			?peer_id,
+			?message_subject,
+			"Starting approval vote import",
+		);
+
 		if let Some(peer_id) = source.peer_id() {
```

Comment:
will not actually work fine, we could start importing the approval vote before the assignment is processed. we also need this checks to be performed before the assignment is processed. so, we would need buffer the approvals until we've processed the assigment.